src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs

```diff
@@ -973,13 +995,15 @@ public static int IndexOfAny(ref char searchSpace, char value0, char value1, cha
                         do
                         {
                             Vector256<ushort> search = LoadVector256(ref searchSpace, offset);
-                            // Note that MoveMask has converted the equal vector elements into a set of bit flags,
-                            // So the bit position in 'matches' corresponds to the element offset.
+                            // We preform the Or at non-Vector level as we are using the maximum number of non-preserved registers,
+                            // and more causes them first to be pushed to stack and then popped on exit to preseve their values.
```

Comment:
Does the prolog/epilog code have critical perf impact? I suggest doing vector-level OR to avoid the long latency `MoveMask`.

Btw, only Windows has callee-saved SIMD registers, so vector-level OR should improve more on Linux.